src/span.h

```diff
@@ -236,11 +237,35 @@ T& SpanPopBack(Span<T>& span)
     return back;
 }
 
+// From C++20 as_bytes and as_writeable_bytes
+template <typename T>
+Span<const std::byte> AsBytes(Span<T> s) noexcept
+{
+    return {reinterpret_cast<const std::byte*>(s.data()), s.size_bytes()};
+}
+template <typename T>
+Span<std::byte> AsWriteableBytes(Span<T> s) noexcept
```

Comment:
This probably needs `std::enable_if_t<std::is_const_v<T>, Span<std::byte>>` as return type, to prevent using this on const spans.

Reply:
`reinterpret_cast` can not be used to remove constness. This will fail with something like:

```
./span.h:249:13: error: reinterpret_cast from 'const unsigned char *' to 'std::byte *' casts away qualifiers
    return {reinterpret_cast<std::byte*>(s.data()), s.size_bytes()};
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
note: in instantiation of function template specialization 'AsWriteableBytes<const unsigned char>' requested here

Reply:
This has been pulled out into https://github.com/bitcoin/bitcoin/pull/23451/files#diff-f9b6e49de18c8395f9f521ead7ac2c688ed91826ebcecea0b7440b6d659c5abcR244 

@sipa Let me know there, if you are against it.